### PR TITLE
Add new fields to FetchOrgResponse model

### PR DESCRIPTION
**Pull request by shiftcontrol-dan**
The FetchOrgResponse model has been expanded with new fields related to organization's SAML settings and domain restrictions. These include url_safe_org_slug, can_setup_saml, is_saml_in_test_mode, domain, domain_autojoin, and domain_restrict. This update improves the granularity and completeness of the data returned by organization fetch requests to include all of the fields that are actually returned from the API.

### CODE REVIEW

**Comment by shiftcontrol-dan** [comment]
I made an additional update to fix the problem reported in #49

**Comment by shiftcontrol-dan** [comment]
@mrmauer for your action.

File: src/models/fetch_org_response.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use serde_json::Value;
 #[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
 pub struct OrgMetadata {
     #[serde(flatten)]
-    metadata: HashMap<String, Value>,
+    pub metadata: HashMap<String, Value>,
```
Comment (mrmauer) [inline src/models/fetch_org_response.rs:18]: this one's a no-brainer

File: src/models/fetch_org_response.rs [hunk 2]
```diff
@@ -37,6 +37,13 @@ pub struct FetchOrgResponse {
...
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub legacy_org_id: Option<String>,
+    pub url_safe_org_slug: String,
+    pub can_setup_saml: bool,
+    pub is_saml_in_test_mode: bool,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub domain: Option<String>,
+    pub domain_autojoin: bool,
+    pub domain_restrict: bool,
 }
```
Comment (mrmauer) [inline src/models/fetch_org_response.rs:47]: So unfortunately, this type is also used under `FetchOrgsResponse` in the `orgs` vec, and the response data there doesn't provide any of these new fields currently. To handle this we'd need `FetchOrgsResponse` to look like
```
FetchOrgsResponse {
    #[serde(rename = "orgs")]
    pub orgs: Vec<crate::models::FetchOrgBasicResponse>,
    ...
}
```
where `FetchOrgBasicResponse` looks how `FetchOrgResponse` does currently.

**Review by mrmauer — COMMENTED** [review]
Thanks again for the PR!
There's one issue with the change to `FetchOrgResponse` that would need addressing before merging (otherwise responses from the org query endpoint will fail to serialize). But if you're blocked on `OrgMetadata.metadata` being public, we can merge that in separately on another PR.

File: src/models/fetch_org_response.rs (hunk 1 above)
Comment (shiftcontrol-dan) [reply src/models/fetch_org_response.rs:18]: Yes, let me pull it out from this PR, I am not blocked on that.

**Comment by shiftcontrol-dan** [comment]
@mrmauer I've updated this to expose 2 structs:

1. Existing `FetchOrgResponse` for the full org response from fetch_org
2. A new `FetchOrgBasicResponse` for the basic response we receive from the API for the query endpoint that then is used in FetchOrgsResponse instead.

Tested locally and works for both queries.

**Comment by andrew-propelauth** [comment]
Thanks for addressing that! We need to do a little testing on our end, will try and get this out next week

**Review by andrew-propelauth — APPROVED** [review]
lgtm, thank you!

**Pull request merged**